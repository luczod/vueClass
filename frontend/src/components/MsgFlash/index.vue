<script setup lang="js"></script>

<script lang="js">
export default {
  name: 'MsgFlash',
  props: ['msg', 'msgClass'],
};
</script>

<template>
  <p class="msg" :class="msgClass" v-if="msg">{{ msg }}</p>
</template>

<style scoped>
.msg {
  max-width: 31rem;
  position: fixed;
  margin: 1.5rem auto;
  padding: 1.5rem;
  text-align: center;
  border: 1px solid #ccc;
  border-radius: 0.5rem;
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
  transition: all 300ms ease-in;
}

.success {
  bottom: 1rem;
  right: 1rem;
  color: #155724;
  background-color: #d4edda;
  border-color: #c3e6cb;
}

.error {
  top: 1rem;
  left: 50%;
  transform: translate(-50%);
  color: #721c24;
  background-color: #f8d7da;
  border-color: #f5c6cb;
}
</style>
