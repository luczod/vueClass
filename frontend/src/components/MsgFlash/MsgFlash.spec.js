import { describe, expect, it } from 'vitest';
import { mount } from '@vue/test-utils';
import MsgFlash from './index.vue';

describe('<MsgFlash />', () => {
  const wrapper = mount(MsgFlash);

  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('asserts correct props are passed', () => {
    const wrapper = mount(MsgFlash, {
      props: {
        msg: 'mensagem de erro',
        msgClass: 'error',
      },
    });

    expect(wrapper.vm.msg).toBe('mensagem de erro');
    expect(wrapper.vm.msgClass).toBe('error');
  });

  it('not render if msg is null', () => {
    const wrapper = mount(MsgFlash);

    expect(wrapper.find('#msg').exists()).toBe(false);
  });

  it('should match rendered', () => {
    const wrapper = mount(MsgFlash, {
      props: {
        msg: 'mensagem de erro',
        msgClass: 'error',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
