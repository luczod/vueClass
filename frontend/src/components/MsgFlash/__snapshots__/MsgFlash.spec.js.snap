// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MsgFlash /> > should match rendered 1`] = `
<p
  class="msg error"
  data-v-2df7bc9a=""
>
  mensagem de erro
</p>
`;
